### PR TITLE
Added avatar unit tests

### PR DESCRIPTION
**Pull request by hyperlogic**
* Added a copy of the [jasmine](http://jasmine.github.io/) test framework to examples/libraries.
* Added a simple set of avatar unit tests to examples/tests.

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: 9ab32552854804cfc763f87baf07d704f6c7a209 available at: https://gist.github.com/a668130bb63d669b0b205d1530e22856<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Alpha-PR7689.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Alpha-PR7689.dmg

File: examples/libraries/jasmine/hifi-boot.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
+
+(function() {
+  function ConsoleReporter(options) {
+    this.jasmineStarted = function (obj) {
```
Comment (ZappoMan) [inline examples/libraries/jasmine/hifi-boot.js]: coding standard says 4 space indents

**Comment by kevinmthomas-carpool** [comment]
## QA Test
**PR Build:** 7689
**Testing Architecture:** 64-bit PC Desktop
**Testing OS:** Windows 10
**Testing Locations:** localhost, playa, cellscience, forum
**Smoke Test Results:** Partially successful test.

                                  |localhost |playa    |cellscience |forum  
----------------------------------|----------|---------|------------|-----
Render Rate:                      |54.02     |55.09    |55.13       |51.17    
Present Rate:                     |60.00     |60.02    |60.00       |60.00   
Simrate:                          |52.96     |53.98    |56.07       |53.15    
Render Stats Num Buffers:         |3360      |2610     |2524        |4196   
Render Stats gpu::Buffer Memory:  |38.7Mb    |62.3Mb   |34.8Mb      |56.4Mb        
Render Stats Num Textures:        |413       |454      |192         |782    
Render Stats gpu::Texture Memory: |1093.8Mb  |1416.6Mb |460.4Mb     |1436.7Mb         
Render Stats Triangles:           |820k      |493k     |509k        |1004k    
Render Stats Drawcalls:           |487       |270      |193         |740   
Render Stats Items:               |214       |241      |794         |344
GPU-Z Memory Size                 |4096Mb    |4096Mb   |4096Mb      |4096Mb
GPU-Z Initial Memory		  |215Mb     |215Mb    |215Mb       |215Mb
GPU-Z Memory Used		  |1593Mb    |2213Mb   |1546Mb      |2223Mb
GPU-Z Memory Load		  |9%        |12%      |9%          |13%

**localhost ‘createtestblocks.js’ Result:** Successful test.
**localhost ATP Upload (kelectricguitar10.fbx):** Successful test.
**localhost ‘harmonicOscillator.js’ Result:** Successful test.
**localhost Echo Server Audio Result:** Successful test.
**localhost ‘reverbTest.js’ Result:** Successful test.
**Additional Test Results Requested:** Partially successful test.  Upon initial load of content on ```localhost``` received error in log ```Physics disabled until entity loads:  QUuid("{8cbbdab6-b1fb-47e1-be96-a7a953d16ca0}") "home_model_deskDrawer_right"```.  to which content was not loading properly.  Attempted to ```Reload content(Cleares all caches)``` and reset ```RAM Caches Size``` cleared the ```Disk Cache Editor``` to which yielded an unsuccessful load of content.  Log file did not provide any related details.  Relogged PR Interface, same behavior occured.  Restarted Console and behaved normally.  Log file included for deeper analysis. 
**Successful Logout/Crash:** Successful test.
[attachments.zip](https://github.com/highfidelity/hifi/files/222805/attachments.zip)

**Comment by hyperlogic** [comment]
About the partially successful smoke test, this PR really should not affect the content loading.  I would verify this occurs on release 4715, which is what this PR is based on, or earlier builds.

**Comment by hifi-gustavo** [comment]
Build console output for commit: efb6a5c3f6360ac26559ec31f232c0d35b4f18d0 available at: https://gist.github.com/a668130bb63d669b0b205d1530e22856<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Alpha-PR7689.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Alpha-PR7689.dmg

**Comment by hyperlogic** [comment]
The build failed downloading tbb for some reason on the linux build.  Lets try this again.

test this please

**Comment by hifi-gustavo** [comment]
Build console output for commit: efb6a5c3f6360ac26559ec31f232c0d35b4f18d0 available at: https://gist.github.com/a668130bb63d669b0b205d1530e22856<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Alpha-PR7689.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Alpha-PR7689.dmg

**Comment by OmegaHeron** [comment]
What Kevin saw with physics disabled warning is likely a thankfully rare for most event where you'll find the hang is actually a texture failing to load for an extended period.  It's a well known thing for me as I have one model that, for me, it's a 100% reproduction case but not so for others.

**Pull request merged**